bin/mdsctl

```diff
@@ -742,42 +775,238 @@ installSimulator() {
+      ${configs[namespace-nats-account-server]:+--namespace ${configs[namespace-nats-account-server]}} \
+      ${configs[values]:+$(helmOptions values ${configs[values]})} \
+      ${configs[values-nats]:+$(helmOptions values ${configs[values-nats]})} \
+      ${configs[presets]:+$(helmOptions set ${configs[presets]})} \
+      ${configs[presets-nats]:+$(helmOptions set ${configs[presets-nats]})} \
+      ${configs[sets]:+$(helmOptions set ${configs[sets]})} \
+      ${configs[sets-nats]:+$(helmOptions set ${configs[sets-nats]})} \
+      $(helmOptions set nscOperatorDirectory=${nscOperatorDirectory})
+      ${configs[setfiles]:+$(helmOptions set-file ${configs[setfiles]})} \
+      ${configs[setfiles-nats]:+$(helmOptions set-file ${configs[setfiles-nats]})} \
+      ${configs[dryrun]:+--dry-run --debug}
+
+    kubectl --namespace ${configs[namespace-nats-account-server]} wait --for=condition=Ready pod/nats-account-server \
+        --timeout=${configs[pause-kubectl]}s || error "nas installation error"
+  }
 }
 
 installNatsbox() {
   check ${FUNCNAME[0]}
 
   [[ "$(helm list --short ${configs[namespace-nats-box]}-nats-box)" = "${configs[namespace-nats-box]}-nats-box" ]] || \
-    helm install ${configs[home]}/helm/nats-box --name ${configs[namespace-nats-box]}-nats-box --namespace ${configs[namespace-nats-box]}
+    helm install ${configs[home]}/helm/nats-box --name ${configs[namespace-nats-box]}-nats-box \
+      ${configs[namespace-nats-box]:+--namespace ${configs[namespace-nats-box]}} \
+      ${configs[values]:+$(helmOptions values ${configs[values]})} \
+      ${configs[values-nats]:+$(helmOptions values ${configs[values-nats]})} \
+      ${configs[presets]:+$(helmOptions set ${configs[presets]})} \
+      ${configs[presets-nats]:+$(helmOptions set ${configs[presets-nats]})} \
+      ${configs[sets]:+$(helmOptions set ${configs[sets]})} \
+      ${configs[sets-nats]:+$(helmOptions set ${configs[sets-nats]})} \
+      ${configs[setfiles]:+$(helmOptions set-file ${configs[setfiles]})} \
+      ${configs[setfiles-nats]:+$(helmOptions set-file ${configs[setfiles-nats]})} \
+      ${configs[dryrun]:+--dry-run --debug}
 }
 
 installStan() {
   check ${FUNCNAME[0]}
 
-  [[ "$(helm list --short ${configs[namespace-stan]}-nats-streaming-server)" = "${configs[namespace-stan]}-nats-streaming-server" ]] || \
-    helm install ${configs[home]}/helm/nats-streaming-server --name ${configs[namespace-stan]}-nats-streaming-server \
-      --namespace ${configs[namespace-stan]}
+  [[ "$(helm list --short ${configs[namespace-nats-stan]}-nats-streaming-server)" = "${configs[namespace-nats-stan]}-nats-streaming-server" ]] || {
+    # todo: make better
+    stanCreds=${HOME}/.nkeys/creds/KO/SYS/sys.creds
+
+    helm install ${configs[home]}/helm/nats-streaming-server --name ${configs[namespace-nats-stan]}-nats-streaming-server \
+      ${configs[namespace-nats-stan]:+--namespace ${configs[namespace-nats-stan]}} \
```

Comment:
I don't know bash very well but is there a way to turn this into a loop?

Reply:
there is clearly a pattern to the helm invocation overlays. as an example, the installSurveyor() function used to invoke the varying helm charts via a loop, only varying the chart name and it's associates. in introducing `kubectl wait` i opted to flatten out the invocations and the relevant kubectl-wait after each operation. this could be wrapped up as a loop again with a trailing switch statement keyed from each helm-name.

so yeah in principle.

back to the above code fragment and if i understand your inquiry correctly, it would be possible to craft a 'helm-invocation-materializer' that could create clean things up a bit given patterns are clearly emerging.

Reply:
Yeah pretty much.